src/rules/relayEdgeTypesSpecPinterestCustomization.ts

```diff
@@ -0,0 +1,186 @@
+import {
+  ArgumentNode,
+  ASTVisitor,
+  DirectiveNode,
+  EnumValueNode,
+  FieldDefinitionNode,
+  ObjectTypeDefinitionNode,
+  TypeDefinitionNode,
+  ValidationContext,
+} from 'graphql';
+
+import { print } from 'graphql/language/printer';
+
+import { getNodeName, unwrapType, ValidationError } from '../utils';
+
+const MANDATORY_FIELDS = ['cursor', 'node'];
+
+export function RelayEdgeTypesSpecPinterestCustomization(
+  context: ValidationContext,
+): ASTVisitor {
+  const hasStringArgument = (argument: ArgumentNode) => {
+    return (argument.value as EnumValueNode).value == 'STRING';
+  };
+
+  const hasDirectiveWithStringArgument = (directive: DirectiveNode) => {
+    const result = directive.arguments?.some(hasStringArgument);
+    return typeof result === 'boolean' ? result : false;
+  };
+
+  const hasStringType = (context: ValidationContext, cursorTypeStr: string) => {
+    const returnsString = cursorTypeStr === 'String';
+    if (!returnsString) {
+      // Checks for scalars that resolve as string as well. Example:
+      // scalar Cursor
+      // @serializationPrimitive(
+      //     type: STRING
+      // )
+      const schemaType = context.getSchema().getType(cursorTypeStr);
+      const hasString = schemaType?.astNode?.directives?.some(
+        hasDirectiveWithStringArgument,
+      );
+      return hasString !== undefined ? hasString : false;
+    }
+    return returnsString;
+  };
+
+  const hasValidNodeType = (
+    context: ValidationContext,
+    node: FieldDefinitionNode,
+  ) => {
+    const allowedTypes = [
+      'UnionTypeDefinition',
+      'ObjectTypeDefinition',
+      'ScalarTypeDefinition',
+      'EnumTypeDefinition',
+      'InterfaceTypeDefinition',
+    ];
+    const nodeFieldType = node?.type;
+    const nodeFieldName = nodeFieldType
+      ? getNodeName(unwrapType(nodeFieldType))
+      : 'undefined';
+    if (!nodeFieldName) {
+      return false;
+    }
+    const schemaType = context.getSchema().getType(nodeFieldName);
+    const implementsNode = allowedTypes.includes(
```

Comment:
why is this called `implementsNode`? This doesn't actually check if the node in an edge `implements Node` does it?

Reply:
You are right. My fault of not changing this naming, as we were trying to check if it implemented Node in a previous implementation but, since that's not required/recommended by the spec, we removed that.